src/store/mod.rs

```diff
@@ -57,7 +57,18 @@ impl Store {
         Ok(result)
     }
 
-    pub fn insert_account(&self, account: &Account) -> Result<(), StoreError> {
+    pub fn insert_account(&mut self, account: &Account) -> Result<(), StoreError> {
+        let tx = self.db.transaction().unwrap();
+
+        Self::insert_account_initial(&tx, account)?;
+        Self::insert_account_code(&tx, account.code())?;
```

Comment:
I synced this fork with the newer commits from the upstream repo and we now have relations on the SQL tables. Because of this, when you create 2 accounts with the same template (for instance, `BasicImmutable`), the account code insertion would fail (because the account code root already exists ont he table) and it would roll back the transaction. We should find a way to insert if it does not currently exist on the table so that the account creation does not fail entirely.

Reply:
You are right, I think we should first check if the primary key (AccountCode root) already exists, and if it does return a `StoreError:AccountAlreadyExists`